example/CMakeLists.txt

```diff
@@ -37,6 +37,9 @@ add_executable(coroutine_echo_client
 )
 add_dependencies(coroutine_echo_client crux)
 target_link_libraries(coroutine_echo_client crux ${EXTRA_LIBS})
+if(MSVC)
+  set_target_properties(coroutine_echo_client PROPERTIES LINK_FLAGS "${LINK_FLAGS} /SAFESEH:NO")
```

Comment:
The coroutine library found by FindBoost is marked as not SEH safe when static linking is turned on, and so fails anything linking to it with SEH on. That FindBoost script is buggy as hell on Windows, it has very broken detection logic whereby it insists on the DLL variant during detection irrespective of whether static link is on or not, but then goes on to use an unstable mix of maybe the DLL import libraries or maybe the static libraries. I wasted a day working around it :(. Besides on Windows because DLL paths aren't hard coded into linked binaries, debugging a shared library linked executable is painful, that's why I turned on static linking only for Windows. Make debugging sane.